### ISSUE TITLE
Update Data Dashboard chart stacking

### ISSUE THREAD

**Issue by christina-luong** [PublicDataWorks/complaint-manager#597]
In the Data Dashboard screen, the chart is stacked (12, 24) making it hard to read and understand the data.

![Image](https://github.com/PublicDataWorks/complaint-manager/assets/167580943/d0ef03cf-26ed-448e-9889-57ff45163547)

### Steps to reproduce
List any steps to reproduce behavior 
1. Step 1
2. Step 2
3. Step 3

### Expected behavior
Describe the expected behavior

### Checklist
- [ ] Indicate bug severity on the right

**Additional context**

**Comment by rowsnr** [comment]
https://hcsoc-data.hawaii.gov/public-data <--- 


![Image](https://github.com/PublicDataWorks/complaint-manager/assets/165818433/b2424a31-4133-4575-b87a-a10de33394ae)

**Comment by rd-huerta** [comment]
### Findings
- we were able to fix the display so that now the "Rule" is showing on the y-axis instead of the "Paragraph" and we were able to fix the labels so that you can read them fully:

![Image](https://github.com/PublicDataWorks/complaint-manager/assets/114006491/0bd99167-17b9-4b72-94a5-b797fa4a709a)

### Other Findings (would require more time/work)
This is another potential option that would require more time:



![Image](https://github.com/PublicDataWorks/complaint-manager/assets/114006491/b00d805f-2448-4759-bf3e-1be78307f215)

### Questions:
1. Which graph is preferred?
2. Should we invest time in implementing this change?
  2a. To separate the data, we would need to write a loop and create new traces. RIght now, we are only plotting 1 trace [as seen in the code here](https://github.com/PublicDataWorks/complaint-manager/blob/3d1e2d8a27bf9ddcc0f1c43272733f3250f0395f/src/client/common/components/Visualization/models/countTop10Allegations.model.js#L107)

**Comment by rowsnr** [comment]
Good work so far! I think the bottom is preferred! It's still showing as stacked, though, I think. But I like the different colors...Let's chat more. @rd-huerta

**Comment by rnair-tw** [comment]
Would like to ensure rule and paragraph are their own line items, does the bottom allow for this?
E.g. the following are all unique lines:
- Rule 1, Para 2
- Rule 1, Para 3
- Rule 3, Para 2

**Comment by heliofreitastw** [comment]
The implementation of the second graph would be more difficult. Also, we could have a really big graph horizontally given that there are several paragraphs in the system.

**Comment by rowsnr** [comment]
would it make sense to make the graph "stack" vertically then similar to how it is now? while obviously "unstacking" each rule/paragraph appropriately. and have users scroll within the box to view? i mean, that's awful but it would work for now.

**Comment by rd-huerta** [comment]
From our perspective, there are a couple options:
**Solution 1:** We have a solution that can be implemented right now. We would need to spend a little more time refactoring the code and writing tests. 

![Image](https://github.com/PublicDataWorks/complaint-manager/assets/114006491/a974df39-fc0c-42a4-8ee9-0ce256bc8703)

**Solution 2:** Unstacking/ungrouping the bars. This is a heavier lift that would require more time investment. We could use the same bar chart shown in the image above, but reduce the size of the bars to accommodate the 10 data points. We aren't sure this is the best approach, however.

**Comment by rnair-tw** [comment]
Yes, as part of this card, would need to ungroup the bars. The users currently need each bar to be displayed separately such as this:

![Image](https://github.com/PublicDataWorks/complaint-manager/assets/91492037/9db5d6c4-6849-43c4-9d25-0f6ae8de369c)

**Comment by rd-huerta** [comment]
Met with @rnair-tw, @rowsnr , and @christina-luong to discuss which graph to use. Waiting on feedback from NOIPM.

**Comment by rnair-tw** [comment]
Clarified with NOIPM, they like to screengrab the chart for reports etc, so does not work for them to have anything stacked (i.e. need to hover over it to see details). Instead, need the following:
- All bars are separate (Rule 1, Para 2; Rule 1, Para 3; Rule 3, Para 2 are all their own line)
- Only display the top 10 bars

**Comment by rd-huerta** [comment]
Still needs some work, but here are the changes we have made so far:



![Image](https://github.com/PublicDataWorks/complaint-manager/assets/114006491/f83ee6fc-c701-4acc-ae98-a9d2ee9f9701)

**Comment by rowsnr** [comment]
I think this is looking good. Any chance we could push the y-axis title further to the left? Also the x-axis title is vague. Is that the number of allegations received? @rd-huerta @rnair-tw

**Comment by rd-huerta** [comment]
### What's been done: 
- adjusted the y-axis title
- changed the x-axis title to "Number of Allegations"

![Image](https://github.com/PublicDataWorks/complaint-manager/assets/114006491/7e155847-bddd-4759-b83c-1d6cd6ed02a5)

### What still needs to be done:
- modify tests to account for the new structure of the graph

### Questions
1. What is directive? Is it needed, or can we use just the rule & paragraph? 
- In the code, we are seeing some examples of `directive || rule`

**Comment by rd-huerta** [comment]
Tests are passing. Tomorrow we will do a bit of refactoring and then a desk check.

**Comment by rd-huerta** [comment]
We are blocked from doing a desk check by tests that are not related to our work failing in the pipeline. Once the pipeline is fixed, our changes can be pushed to the CI environment.

**Comment by heliofreitastw** [comment]
I talked with @rowsnr we decided to limit the paragraph text on the left size of the graph to 40 characters and include the ellipsis (`...`) at the end when this happens. If the paragraph is smaller than 40 characters there is not problem. Also, she suggested abbreviating paragraph to `par`. Finally, the hover text will be added back with the full paragraph text.

**Comment by heliofreitastw** [comment]
The allegations chart in the [data page] (https://noipm-production.herokuapp.com/data) is cutting the name of the rule + paragraph displayed on the left side of the chart.

**Comment by christina-luong** [comment]
@heliofreitastw Received some feedback on the Top Allegations chart in prod. The bars appear to be merging likely due to lack of space. Currently exploring the possibility of limiting Top Allegation to Top 5 or Top 8, but pending feedback from Christian.

![Image](https://github.com/PublicDataWorks/complaint-manager/assets/167580943/9dee98de-50fc-4bbc-8dfd-7141d4df0c80)

**Comment by christina-luong** [comment]
Christian would prefer that we limit to Top 8 Allegations (or more).

**Comment by thoughtworks-vwong** [comment]
Verified top allegations chart is showing top 8 allegations, and they are no longer grouped together and hard to read.

![Image](https://github.com/PublicDataWorks/complaint-manager/assets/134408135/190da570-0983-480e-ba2a-8646f730c385)
